api/types/desktop_test.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWindowsDesktopsSorter(t *testing.T) {
+	t.Parallel()
+
+	testValsUnordered := []string{"d", "b", "a", "c"}
+
+	makeDesktops := func(testVals []string, testField string) []WindowsDesktop {
+		desktops := make([]WindowsDesktop, len(testVals))
+		for i := 0; i < len(testVals); i++ {
+			testVal := testVals[i]
+			var err error
+			desktops[i], err = NewWindowsDesktopV3(
+				getTestVal(testField == ResourceMetadataName, testVal),
+				nil,
+				WindowsDesktopSpecV3{
+					Addr: getTestVal(testField == ResourceSpecAddr, testVal),
+				})
+
+			require.NoError(t, err)
+		}
+		return desktops
+	}
+
+	cases := []struct {
+		name      string
+		fieldName string
+	}{
+		{
+			name:      "by name",
+			fieldName: ResourceMetadataName,
+		},
+		{
+			name:      "by addr",
+			fieldName: ResourceSpecAddr,
+		},
+	}
+
+	for _, c := range cases {
+		c := c
+		t.Run(fmt.Sprintf("%s desc", c.name), func(t *testing.T) {
+			t.Parallel()
```

Comment:
You can remove the `t.Parallel()` from the subtests here. You only have two cases, and they are sorting tiny lists of 4 elements, so it's probably faster just to run them sequentially.